addons/binding/org.openhab.binding.mqtt.generic/src/main/java/org/openhab/binding/mqtt/generic/internal/convention/homeassistant/HandlerConfiguration.java

```diff
@@ -27,9 +29,31 @@
     /**
      * The MQTT prefix topic
      */
-    public String basetopic = "homeassistant";
+    private String basetopic = "homeassistant";
     /**
      * The object id. This is comparable to a Homie Device.
      */
-    public String objectid = "";
+    private String objectid = "";
+
+    public HandlerConfiguration() {
+    }
+
+    public HandlerConfiguration(String basetopic, String objectid) {
+        super();
+        this.basetopic = basetopic;
+        this.objectid = objectid;
+    }
+
+    public String getBasetopic() {
+        return basetopic;
+    }
+
+    public String getObjectid() {
+        return objectid;
+    }
+
+    public void toProperties(Map<String, Object> properties) {
```

Comment:
This is an anti-pattern (using an argument as output).
Please call this method "appendToProperties" or alike and return the map again. And add javadoc please.